File: backend/src/main/java/com/corusoft/ticketmanager/tickets/controllers/dtos/ParsedTicketDTO.java
package com.corusoft.ticketmanager.tickets.controllers.dtos;

import jakarta.validation.constraints.*;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ParsedTicketDTO {
    private Long id;

    @PastOrPresent
    private LocalDateTime registeredAt;

    @NotBlank
    private String supplier;

    @NotBlank
    private String category;

    @NotBlank
    private String subcategory;

    @PastOrPresent
    private LocalDate emmitedAtDate;

    @NotBlank
    private String emmitedAtTime;

    @NotBlank
    private String country;

    @NotBlank
    private String currency;

    @PositiveOrZero
    private Float totalTax;

    @PositiveOrZero
    private Float totalAmount;
}

File: mobile/app/src/main/java/com/corusoft/ticketmanager/backend/dtos/tickets/ParsedTicketDTO.kt
package com.corusoft.ticketmanager.backend.dtos.tickets

import java.time.LocalDate
import java.time.LocalDateTime

data class ParsedTicketDTO(
    val id: Long?,
    val registeredAt: LocalDateTime?,
    val supplier: String?,
    val category: String?,
    val subcategory: String?,
    val emmitedAtDate: LocalDate?,
    val emmitedAtTime: String?,
    val country: String?,
    val currency: String?,
    val totalTax: Float?,
    val totalAmount: Float?
)
